brave/brave-core
# Add advertiser split testing for Brave ads

Resolves https://github.com/brave/brave-browser/issues/14613

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
Run with `npm start -- --enable-logging=stderr --vmodule="*/variations/*"=6,"*/bat-native-ads/*"=6,"*/brave_ads/*"=6,"*/brave_user_model/*"=6,"*/bat_ads/*"=6 --use-dev-goupdater-url --brave-ads-staging --rewards=staging=true --variations-server-url=https://variations.bravesoftware.com/seed --fake-variations-channel=beta`

- Inject test catalog that has a creative set with field `"splitTestGroup": "GroupA"` set.
- Start browser without enabling ads and quit. Then start a second time with enabling ads (**This is important since `seed` will only be initialised on restart!**)
- Verify in logs that study loaded correctly with `[10210:775:0313/084147.162484:VERBOSE1:study_filtering.cc(346)] Kept study AdvertiserSplitTestingStudy.`
- trigger ad and verify it is being served and not filtered out.
- Change the test catalog to contain the field `"splitTestGroup": "GroupB"` and wipe profile.
- Re-run test and verify the frequency cap is now excluding this ad.
